### ISSUE TITLE
Requirement Handling in written flow.

### ISSUE THREAD

**Issue by DhrutiShah** [Promact/requirement-management#7]
Client is creating the requirements in red mine with just basic information. but what should be done in that feature in DETAIL will always discussed on call. this is what they prefers. so what happens is we have to draft the mail for the same for written proof. so first of all have to draft just understanding flow as per discussion. then team members analyzed the tasks and mail iteration started for queries and so on..

Now if there are 3 to 4 points which we have discussed on call, then he/she will confirmed that points. but never confirmed the full understanding flow, because they have to read such a long mail. and moreover features are too small, so we can't create wire frames for each feature by which he can easily confirmed the understanding flow. but sometimes it happens that what he is trying to say and what we are understanding is might be different. so after full flow implementation and testing, at that time of demo they asked we don't want this. we want this. so the rework comes. 

so what approach can be followed by which they also confirmed the understanding flow and also mail iteration can be reduced?

**Comment by RiyaShah** [comment]
I presume it's more of an issue with the client than an issue with requirement management. The reason being that if the client does not confirm the full understanding flow just because he/she has to read long email then I guess we can't do anything in our system too. 
I suggest that there should be a deal with the client that only those features will be implemented whose flow has been confirmed. This way there won't be any conflict to do rework. I also suggest sending the client the wireframes too while sending the flow of the features wherever applicable.

**Comment by karandesai28** [comment]
I tend to have different point of view @RiyaShah mam. I'll try to re-frame here what I think @DhrutiShah  mam wanted to convey and how we can inculcate the solution to issue in our system:

There is no mechanism to convey client that his requirement specification is incomplete, unclear, shallow or detrimental. This increases heavy consumption of time and resources in asking the same in numerous email threads, and analyzing, maintaining and updating estimation after each reply. If somehow, in our system we inculcate phases like- (again leading to [this](https://github.com/Promact/requirement-management/issues/9) )

1. "Requirement's Analysis in progress"
2.  "Waiting for client's reply"
3. "Incomplete requirements" (Puts the functionality on halt indicating client to add more details)
4. "Waiting for confirmation"
5. "Ready for implementation"

This way we can convey the clients from the system itself that developers would or would not be able to proceed based on requirements specified. This will automatically emphasize client to put relevant and all required details beforehand which will reduce query email threads, rework and time consumed in analysis.

This will also resolve client's problem of reading long thread mails as they can comment directly in the system asking what more details we want or what issue is being faced.

**Comment by RiyaShah** [comment]
Yes incorporating the above phases in our system would be a good idea which will be a systematic approach from requirement analysis to its implementation phase. 
However, the problem in your case is that "client doesn't confirm the full understanding flow" won't be resolved completely. The flow has to be sent to the client either via email or using this system and client will have to read the entire flow anyhow. Ultimately, it's up to the client to confirm it. 
Of course, the further questionnaires on the flow can be done via our system.
